ts/screens/authentication/IdpLoginScreen.tsx

```diff
@@ -144,9 +134,14 @@ const IdpLoginScreen = (props: Props) => {
     undefined
   );
 
-  const loginSource = useLollipopLoginSource({
-    loggedOutWithIdpAuth: props.loggedOutWithIdpAuth
-  });
+  const loginUri = pipe(
```

Comment:
In this case, I don't see an improvement in using `fp-ts` to just combine native types into an `Option` for later outputting a native-again type (we are just mixing things here). 
Either we deal directly with the native type without `fp-ts` (chaining optionals and handling undefined) or we return an `Option` type from the `fp-ts` `pipe` and change `useLollipopLoginSource` signature to deal with an `Option<string>` on input instead of a `string`.
What are your thoughts on this?

Reply:
Yup, I agree.